test/Entity/TimestampableTraitTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-authentication-oauth2 for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-authentication-oauth2/blob/master/LICENSE.md
+ *     New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace ZendTest\Expressive\Authentication\OAuth2\Entity;
+
+use DateTime;
```

Comment:
Quick question: should we be using `DateTime`, or `DateTimeImmutable`? Which would make more sense for the domain?

Reply:
I would say accept `DateTimeInterface`, convert to `DateTimeImmutable` if mutable.

Reply:
I used `DateTimeInterface` as parameters of  `TimestampableTrait` and `DateTimeImmutable` in `TimestampableTrait::timestampOnCreate()` function.